### ISSUE TITLE
r2dec installation failed

### ISSUE THREAD

**Issue by dindibo** [wargio/r2dec-js#147]
**Describe the bug**
I've cloned the repo to my computer and then I did the following:

$r2pm init
$r2pm install r2dec

the second command failed and it printed: 
`duktape/duk_console.c:13:10: fatal error: r_cons.h: No such file or directory
 #include <r_cons.h>
          ^~~~~~~~~~
compilation terminated.
core_pdd.c:8:10: fatal error: r_types.h: No such file or directory
 #include <r_types.h>
          ^~~~~~~~~~~`

then I did `sudo apt install pkg-config`
and different kind of message appeared:

`core_pdd.c:273:2: error: unknown type name ‘RCoreAutocomplete’
  RCoreAutocomplete *pdd = r_core_autocomplete_add (core->autocomplete, "pdd", R_CORE_AUTOCMPLT_DFLT, true);
  ^~~~~~~~~~~~~~~~~
core_pdd.c:273:27: warning: implicit declaration of function ‘r_core_autocomplete_add’; did you mean ‘r_core_cmpwatch_add’? [-Wimplicit-function-declaration]
  RCoreAutocomplete *pdd = r_core_autocomplete_add (core->autocomplete, "pdd", R_CORE_AUTOCMPLT_DFLT, true);
                           ^~~~~~~~~~~~~~~~~~~~~~~
                           r_core_cmpwatch_add
core_pdd.c:273:56: error: ‘RCore {aka struct r_core_t}’ has no member named ‘autocomplete’
  RCoreAutocomplete *pdd = r_core_autocomplete_add (core->autocomplete, "pdd", R_CORE_AUTOCMPLT_DFLT, true);
                                                        ^~
core_pdd.c:273:79: error: ‘R_CORE_AUTOCMPLT_DFLT’ undeclared (first use in this function); did you mean ‘R_CORE_CMD_DEPTH’?
  RCoreAutocomplete *pdd = r_core_autocomplete_add (core->autocomplete, "pdd", R_CORE_AUTOCMPLT_DFLT, true);
                                                                               ^~~~~~~~~~~~~~~~~~~~~
                                                                               R_CORE_CMD_DEPTH
core_pdd.c:273:79: note: each undeclared identifier is reported only once for each function it appears in
core_pdd.c:274:31: error: ‘RCore {aka struct r_core_t}’ has no member named ‘autocomplete’
  r_core_autocomplete_add (core->autocomplete, "pdd?", R_CORE_AUTOCMPLT_DFLT, true);
                               ^~
core_pdd.c:275:31: error: ‘RCore {aka struct r_core_t}’ has no member named ‘autocomplete’
  r_core_autocomplete_add (core->autocomplete, "pdd*", R_CORE_AUTOCMPLT_DFLT, true);
                               ^~
core_pdd.c:276:31: error: ‘RCore {aka struct r_core_t}’ has no member named ‘autocomplete’
  r_core_autocomplete_add (core->autocomplete, "pdda", R_CORE_AUTOCMPLT_DFLT, true);
                               ^~
core_pdd.c:277:31: error: ‘RCore {aka struct r_core_t}’ has no member named ‘autocomplete’
  r_core_autocomplete_add (core->autocomplete, "pddb", R_CORE_AUTOCMPLT_DFLT, true);
                               ^~
core_pdd.c:278:31: error: ‘RCore {aka struct r_core_t}’ has no member named ‘autocomplete’
  r_core_autocomplete_add (core->autocomplete, "pddi", R_CORE_AUTOCMPLT_DFLT, true);
                               ^~
core_pdd.c:279:31: error: ‘RCore {aka struct r_core_t}’ has no member named ‘autocomplete’
  r_core_autocomplete_add (core->autocomplete, "pdds", R_CORE_AUTOCMPLT_DFLT, true);
                               ^~
core_pdd.c:280:31: error: ‘RCore {aka struct r_core_t}’ has no member named ‘autocomplete’
  r_core_autocomplete_add (core->autocomplete, "pddu", R_CORE_AUTOCMPLT_DFLT, true);
                               ^~
core_pdd.c:281:48: error: ‘R_CORE_AUTOCMPLT_OPTN’ undeclared (first use in this function); did you mean ‘R_CORE_AUTOCMPLT_DFLT’?
  r_core_autocomplete_add (pdd, "--as-comment", R_CORE_AUTOCMPLT_OPTN, true);
                                                ^~~~~~~~~~~~~~~~~~~~~
                                                R_CORE_AUTOCMPLT_DFLT
`

**Component**
- [X] core
- [ ] arm
- [ ] avr
- [ ] m68k
- [ ] mips
- [ ] ppc
- [ ] sparc
- [ ] v850
- [ ] wasm
- [ ] x86-64

**Comment by radare** [comment]
You need to install r2 in order to build r2dec

Let me do a blind guess here. You are using debian and have installed only the non-dev pkg

**Comment by wargio** [comment]
You don't need to clone anything.
Seems that you are using an obsolete version of r2.

Use the git version.

**Comment by dindibo** [comment]
> You don't need to clone anything.
> Seems that you are using an obsolete version of r2.
> 
> Use the git version.

Thanks, I just installed radare2 repository instead of the version from apt install and now it works.

**Comment by n4rfy** [comment]
> **Describe the bug**
> I've cloned the repo to my computer and then I did the following:
> 
> $r2pm init
> $r2pm install r2dec
> 
> the second command failed and it printed:
> `duktape/duk_console.c:13:10: fatal error: r_cons.h: No such file or directory #include <r_cons.h> ^~~~~~~~~~ compilation terminated. core_pdd.c:8:10: fatal error: r_types.h: No such file or directory #include <r_types.h> ^~~~~~~~~~~`
> 
> then I did `sudo apt install pkg-config`
> and different kind of message appeared:
> 
> `core_pdd.c:273:2: error: unknown type name ‘RCoreAutocomplete’ RCoreAutocomplete *pdd = r_core_autocomplete_add (core->autocomplete, "pdd", R_CORE_AUTOCMPLT_DFLT, true); ^~~~~~~~~~~~~~~~~ core_pdd.c:273:27: warning: implicit declaration of function ‘r_core_autocomplete_add’; did you mean ‘r_core_cmpwatch_add’? [-Wimplicit-function-declaration] RCoreAutocomplete *pdd = r_core_autocomplete_add (core->autocomplete, "pdd", R_CORE_AUTOCMPLT_DFLT, true); ^~~~~~~~~~~~~~~~~~~~~~~ r_core_cmpwatch_add core_pdd.c:273:56: error: ‘RCore {aka struct r_core_t}’ has no member named ‘autocomplete’ RCoreAutocomplete *pdd = r_core_autocomplete_add (core->autocomplete, "pdd", R_CORE_AUTOCMPLT_DFLT, true); ^~ core_pdd.c:273:79: error: ‘R_CORE_AUTOCMPLT_DFLT’ undeclared (first use in this function); did you mean ‘R_CORE_CMD_DEPTH’? RCoreAutocomplete *pdd = r_core_autocomplete_add (core->autocomplete, "pdd", R_CORE_AUTOCMPLT_DFLT, true); ^~~~~~~~~~~~~~~~~~~~~ R_CORE_CMD_DEPTH core_pdd.c:273:79: note: each undeclared identifier is reported only once for each function it appears in core_pdd.c:274:31: error: ‘RCore {aka struct r_core_t}’ has no member named ‘autocomplete’ r_core_autocomplete_add (core->autocomplete, "pdd?", R_CORE_AUTOCMPLT_DFLT, true); ^~ core_pdd.c:275:31: error: ‘RCore {aka struct r_core_t}’ has no member named ‘autocomplete’ r_core_autocomplete_add (core->autocomplete, "pdd*", R_CORE_AUTOCMPLT_DFLT, true); ^~ core_pdd.c:276:31: error: ‘RCore {aka struct r_core_t}’ has no member named ‘autocomplete’ r_core_autocomplete_add (core->autocomplete, "pdda", R_CORE_AUTOCMPLT_DFLT, true); ^~ core_pdd.c:277:31: error: ‘RCore {aka struct r_core_t}’ has no member named ‘autocomplete’ r_core_autocomplete_add (core->autocomplete, "pddb", R_CORE_AUTOCMPLT_DFLT, true); ^~ core_pdd.c:278:31: error: ‘RCore {aka struct r_core_t}’ has no member named ‘autocomplete’ r_core_autocomplete_add (core->autocomplete, "pddi", R_CORE_AUTOCMPLT_DFLT, true); ^~ core_pdd.c:279:31: error: ‘RCore {aka struct r_core_t}’ has no member named ‘autocomplete’ r_core_autocomplete_add (core->autocomplete, "pdds", R_CORE_AUTOCMPLT_DFLT, true); ^~ core_pdd.c:280:31: error: ‘RCore {aka struct r_core_t}’ has no member named ‘autocomplete’ r_core_autocomplete_add (core->autocomplete, "pddu", R_CORE_AUTOCMPLT_DFLT, true); ^~ core_pdd.c:281:48: error: ‘R_CORE_AUTOCMPLT_OPTN’ undeclared (first use in this function); did you mean ‘R_CORE_AUTOCMPLT_DFLT’? r_core_autocomplete_add (pdd, "--as-comment", R_CORE_AUTOCMPLT_OPTN, true); ^~~~~~~~~~~~~~~~~~~~~ R_CORE_AUTOCMPLT_DFLT `
> 
> **Component**
> 
> * [x]  core
> * [ ]  arm
> * [ ]  avr
> * [ ]  m68k
> * [ ]  mips
> * [ ]  ppc
> * [ ]  sparc
> * [ ]  v850
> * [ ]  wasm
> * [ ]  x86-64

liblz4-dev libradare2-4.3.1 libradare2-common libradare2-dev libuv1-dev radare2

Try installing those dependencies

**Comment by Maijin** [comment]
@n4rfy please open a new issue, the original issue is from march 2019.
